### ISSUE TITLE
 Major updates needed to be maintained

### ISSUE THREAD

**Issue by Venom-Sec** [redcanari/armitage#1]
Hello Redcanari. I'm an armitage fanatic who's been using it for years. I've been following your excellent work and wanted to express my gratitude for your efforts. The reason I'm contacting you is because we discovered an issue that we couldn't publish to github because the feature is disabled. The issue we're discussing is one of armitage. 
 
**Issues**
1) The stageless and staged MSF payload modules aren't recognized from armitage and it hangs when I either execute commands or taking screenshot.
 
2) The network GUI graph needs to be updated to replace the lib/jgraphx.jar to the Java files of the following source that cobalt strike was using it. 

What I was trying to do is look at the libraries of where it came from and how it was executed. I'm deleting it on purpose to see what areas are needed to improve.

**Links**
https://github.com/jgraph/jgraphx
https://jgraph.github.io/mxgraph/

**Images**
![Stageless Listener](https://i.ibb.co/RYXQ8V8/2.png)
![Unrecognized Stageless Listener](https://i.ibb.co/2vB4Kdh/1.png)

**Issues**
Now the second the issue I had to look into is related to GUI of network graph. This is what I had to find out of with the following commands that i had to remove jgraph.jar on purpose

```
cp /opt/armitage/lib/jgraphx.jar .
sudo rm /opt/armitage/lib/jgraphx.jar
$ sudo ./package.sh --debug             
+ ./gradlew assemble
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :armitage:compileJava
/opt/armitage/armitage/src/main/java/graph/CircleLayout.java:6: error: package com.mxgraph.model does not exist
import com.mxgraph.model.mxIGraphModel;
                        ^
/opt/armitage/armitage/src/main/java/graph/CircleLayout.java:7: error: package com.mxgraph.util does not exist
import com.mxgraph.util.mxRectangle;
                       ^
/opt/armitage/armitage/src/main/java/graph/CircleLayout.java:8: error: package com.mxgraph.view does not exist
import com.mxgraph.view.mxGraph;
                       ^
/opt/armitage/armitage/src/main/java/graph/CircleLayout.java:11: error: cannot find symbol
public class CircleLayout extends mxCircleLayout
                                  ^
  symbol: class mxCircleLayout
/opt/armitage/armitage/src/main/java/graph/CircleLayout.java:13: error: cannot find symbol
        public CircleLayout(mxGraph graph, double r) {
                            ^
  symbol:   class mxGraph
  location: class CircleLayout
/opt/armitage/armitage/src/main/java/graph/CircleLayout.java:9: error: package com.mxgraph.layout does not exist
import com.mxgraph.layout.*;
..[snip]..
```

**Comment by allfro** [comment]
Please provide reproduction steps. I can't troubleshoot without valid repro steps.

**Comment by Venom-Sec** [comment]
How do I debug it then?

**Comment by Venom-Sec** [comment]
I'm not an expert in java, all I did was:
$ handler -p windows/meterpreter_reverse_http -L 192.168.56.101 -P 80
and I've selected the metepreter session and typed commands like:
meterpreter > sysinfo
but it hangs up like 10-20 seconds I think then it produces the output

**Comment by allfro** [comment]
Sounds like EDR/AV is killing it

**Comment by Venom-Sec** [comment]
No you're wrong. I had to test to see if armitage recognizes the new payloads especially the stageless ones. I'm testing it from a blackbox perspective to see what payload modules that it recognizes with the newer metasploit modules.

**Comment by allfro** [comment]
The problem is I don't know the sequence of steps you are using to trigger this bug. I just need a video or a list of things that are being performed so I can assist. Otherwise, I'm stuck guessing.

**Comment by Venom-Sec** [comment]
The video has been shut off until the very end. Finally, I choose a session with which I wanted to interact. It will produce an error message stating that it does not recognise the session; I have exhausted my attempts to explain. msf6 > sessions -i 1
![Video](https://media.giphy.com/media/4Kd7jH31tbaNc5oGQA/giphy.gif)

**Comment by allfro** [comment]
Right. so `sessions -i <id>` won't work in Armitage. It never has afaik. What will work is if you right click on the host that shows up and select the session from the context menu. Remember the shell you see is not an interactive shell. It is using msfrpc in the background to interact with the msf core. So when you create a new session it is being handled by a separate thread and therefore requires a separate request to interact with it.

TLDR: use the context menu on the compromised host to interact with the meterpreter shell.

**Comment by Venom-Sec** [comment]
Thanks for Information
